CARET_trace/src/ros_trace_points.cpp

```diff
@@ -1220,6 +1220,31 @@ void ros_trace_rmw_take(
   }
 }
 
+#ifdef ROS_DISTRO_JAZZY
+void ros_trace_rmw_publish(
+  const void * rmw_publisher_handle,
+  const void * message,
+  int64_t init_timestamp
+)
+{
+  static auto & context = Singleton<Context>::get_instance();
+  static auto & controller = context.get_controller();
+
+  if (!controller.is_allowed_process()) {
+    return;
+  }
+
+  if (trace_filter_is_rcl_publish_recorded) {
```

Comment:
The following is one of the suggestions for implementation improvement.
I don't think it's necessary to include this in the current PR, but I'll leave it as text for reference.

---

Until now, we've been determining whether to apply the filter for rmw_publish (=dds_write) based on the presence or absence of rcl_publish records.
However, from jazzy onwards, we can obtain rmw_publish.
Using rmw_publish for filtering might enable a more appropriate implementation.

Reply:
no particular corrections made to this issue